src/marqo/core/semi_structured_vespa_index/semi_structured_add_document_handler.py

```diff
@@ -0,0 +1,74 @@
+from typing import Dict, Any
+
+from marqo.config import Config
+from marqo.core import constants
+from marqo.core.constants import MARQO_DOC_ID
+from marqo.core.document.models.add_docs_params import AddDocsParams
+from marqo.core.models.marqo_index import SemiStructuredMarqoIndex, Field, FieldType, FieldFeature, TensorField
+from marqo.core.semi_structured_vespa_index.semi_structured_vespa_index import SemiStructuredVespaIndex
+from marqo.core.semi_structured_vespa_index.semi_structured_vespa_schema import SemiStructuredVespaSchema
+from marqo.core.unstructured_vespa_index.unstructured_add_document_handler import UnstructuredAddDocumentsHandler
+from marqo.tensor_search import index_meta_cache
+from marqo.tensor_search.telemetry import RequestMetricsStore
+from marqo.vespa.models import VespaDocument
+
+
+class SemiStructuredAddDocumentsHandler(UnstructuredAddDocumentsHandler):
+    def __init__(self, marqo_index: SemiStructuredMarqoIndex, config: Config, add_docs_params: AddDocsParams):
+        super().__init__(marqo_index, config, add_docs_params)
+        self.marqo_index = marqo_index
+        self.vespa_index = SemiStructuredVespaIndex(marqo_index)
+        self.should_update_index = False
+
+    def handle_field(self, marqo_doc, field_name, field_content):
+        self._validate_field(field_name, field_content)
+        text_field_type = self._infer_field_type(field_content)
+        content = self.tensor_fields_container.collect(marqo_doc[MARQO_DOC_ID], field_name,
+                                                       field_content, text_field_type)
+        marqo_doc[field_name] = content
+
+        if isinstance(content, str):
+            # Add missing lexical fields to marqo index
+            if field_name not in self.marqo_index.field_map:
+                self.marqo_index.lexical_fields.append(
+                    Field(name=field_name, type=FieldType.Text,
```

Comment:
Do we have a configurable limit on max tensor and lexical fields?

Reply:
No we don't have one yet. what's in your mind would be a good limit?

Reply:
I think we need some generous limit to catch the odd user who is creating fields in a loop. Perhaps 10 each? Or 20

Reply:
Do you mean we need a limit for each batch or overall? For each batch, it should be fine, since it takes about the same time to add 1 field and 10 fields. We haven't done the experiment to check the max tensor fields Vespa supports, so we will need to pick a random number as an overall limit (or maybe leave it as unlimited to see until we have some extreme use cases which breaks Vespa).

Reply:
Here's the result of my local test:
# adding 2 fields, 'add_documents.update_index': 2159.5356249999895
# adding 10 fields, 2 at a time, first: 2s, following: 6s, 5s, 7s, 5s (there seem to be a cooling down period in Vespa convergence, if adding a 5s sleep in-between, time will reduce to 2s)
# adding 10 fields, 'add_documents.update_index': 2061.9370840000784
# adding 20 fields, 'add_documents.update_index': 1923.3983750000334
# adding 50 fields, 'add_documents.update_index': 2128.181333000043

Reply:
Added the following two env vars with default value set to 100
`MARQO_MAX_TENSOR_FIELD_COUNT_UNSTRUCTURED`
`MARQO_MAX_LEXICAL_FIELD_COUNT_UNSTRUCTURED`